src/common/DecayCounter.cc

```diff
@@ -19,7 +19,6 @@
 
 void DecayCounter::encode(bufferlist& bl) const
 {
-  decay();
```

Comment:
I'm not really agreeing with this change. Is this addressing some problem you came across?

Reply:
> I'm not really agreeing with this change.

could you please help me understand why?

> Is this addressing some problem you came across?

@batrick yes. to be specific, `check-generated.sh` compare following 4 json outputs:

1. `dump_json`
1. `encode decode dump_json`
1. `copy dump_json`
1. `copy_ctor dump_json`

and it consider it a failure if they are not identical. in the case of `DecayCounter `, `dump_json` just calls `DecayCounter::dump()`. so, if we encode a sample of `DecayCounter `,  decode, and then dump it. the output will be different than the output of the original copy. as the first one is decayed twice.

i put the rationale in the commit message also.

Reply:
thanks @batrick . i think the test is still valuable. it make sure we don't break the backward compatibility. instead i changed the `DecayRate` to zero when preparing the samples for testing.